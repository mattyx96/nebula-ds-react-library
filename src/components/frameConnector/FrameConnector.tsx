import {lightJsTokens} from "nebula-ds-tokens"

type BridgeProps = {
  className?: string
}

type NodeProps = {
  fill?: string
  className?: string
  hidden?: boolean
}

export type Props = {
  divider?: boolean
  vertical?: boolean
  className?: string
  bridge?: BridgeProps
  firstNode?: NodeProps
  secondNode?: NodeProps
}

export const FrameConnector = (props: Props) => {
  return (
    <div className={`flex ${props.vertical ? 'flex-col w-fit' : 'items-end flex-1'} ${props.className}`}>
      {props.firstNode?.hidden || props.divider ? <></> :
        <Node fill={props.firstNode?.fill || lightJsTokens.nbFrameBackgroundPrimary}
              className={`${props.vertical ? 'rotate-90 origin-center' : ''} scale-x-[-1] ${props.firstNode?.className}`}/>}
      <Bridge className={props.bridge?.className} vertical={!!props.vertical}/>
      {props.secondNode?.hidden || props.divider ? <></> :
        <Node fill={props.secondNode?.fill || lightJsTokens.nbFrameBackgroundPrimary}
              className={`${props.vertical ? 'rotate-90 origin-center' : ''} ${props.secondNode?.className}`}/>}
    </div>
  )
}

const Node = (props: { fill: string; className?: string }) => (
  <div className={`flex items-center justify-center ` + props.className}>
    <svg width="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path fillRule="evenodd" clipRule="evenodd"
            d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
            fill={props.fill}/>
    </svg>
  </div>
);

const Bridge = (props: { className?: string, vertical: boolean }) => (
  <div
    className={`${props.vertical ? 'h-full w-[6px]' : 'w-full h-[6px]'} bg-frame-background-primaryu ${props.className}`}/>
);
