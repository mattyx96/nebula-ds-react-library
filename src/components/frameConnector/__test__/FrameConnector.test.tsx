import {describe, expect} from 'vitest';
import {it} from 'vitest';
import {FrameConnector} from '../FrameConnector';
import {render} from "@testing-library/react";

describe('FrameConnector', () => {
  it('renders correctly', () => {
    const {container} = render(<FrameConnector/>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with className prop', () => {
    const {container} = render(<FrameConnector className="custom-class"/>);
    expect(container.firstChild).toHaveClass('custom-class');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with bridge className prop', () => {
    const {container} = render(<FrameConnector bridge={{className: 'bridge-class'}}/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('.bridge-class')).toBeInTheDocument();
  });

  it('renders correctly with firstNode props', () => {
    const firstNodeProps = {fill: 'red', className: 'first-node-class'};
    const {container} = render(<FrameConnector firstNode={firstNodeProps}/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('.first-node-class')).toBeInTheDocument();
  });

  it('renders correctly with secondNode props', () => {
    const secondNodeProps = {fill: 'blue', className: 'second-node-class'};
    const {container} = render(<FrameConnector secondNode={secondNodeProps}/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('.second-node-class')).toBeInTheDocument();
  });

  it('hides nodes correctly with hidden prop', () => {
    const {container} = render(<FrameConnector firstNode={{hidden: true}} secondNode={{hidden: true}}/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('svg')).toBeNull();
  });

  it('handles divider prop correctly', () => {
    const {container} = render(<FrameConnector divider/>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('svg')).toBeNull();
  });

  it('renders correctly with all props mixed', () => {
    const props = {
      className: 'custom-class',
      bridge: {className: 'bridge-class'},
      firstNode: {fill: 'red', className: 'first-node-class'},
      secondNode: {fill: 'blue', className: 'second-node-class'},
    };
    const {container} = render(<FrameConnector {...props} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('.custom-class')).toBeInTheDocument();
    expect(container.querySelector('.bridge-class')).toBeInTheDocument();
    expect(container.querySelector('.first-node-class')).toBeInTheDocument();
    expect(container.querySelector('.second-node-class')).toBeInTheDocument();
  });
});
