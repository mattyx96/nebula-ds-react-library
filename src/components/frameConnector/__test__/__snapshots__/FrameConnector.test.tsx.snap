// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FrameConnector > handles divider prop correctly 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="w-full h-[6px] bg-background-accent-200 undefined"
  />
</div>
`;

exports[`FrameConnector > hides nodes correctly with hidden prop 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="w-full h-[6px] bg-background-accent-200 undefined"
  />
</div>
`;

exports[`FrameConnector > renders correctly 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="flex items-center justify-center  scale-x-[-1] undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="w-full h-[6px] bg-background-accent-200 undefined"
  />
  <div
    class="flex items-center justify-center  undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with all props mixed 1`] = `
<div
  class="flex items-end flex-1 custom-class"
>
  <div
    class="flex items-center justify-center  scale-x-[-1] first-node-class"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="red"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="w-full h-[6px] bg-background-accent-200 bridge-class"
  />
  <div
    class="flex items-center justify-center  second-node-class"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="blue"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with bridge className prop 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="flex items-center justify-center  scale-x-[-1] undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="w-full h-[6px] bg-background-accent-200 bridge-class"
  />
  <div
    class="flex items-center justify-center  undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with className prop 1`] = `
<div>
  <div
    class="flex items-end flex-1 custom-class"
  >
    <div
      class="flex items-center justify-center  scale-x-[-1] undefined"
    >
      <svg
        fill="none"
        viewBox="0 0 60 60"
        width="60"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
          fill="#d0cac4"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="w-full h-[6px] bg-background-accent-200 undefined"
    />
    <div
      class="flex items-center justify-center  undefined"
    >
      <svg
        fill="none"
        viewBox="0 0 60 60"
        width="60"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
          fill="#d0cac4"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with firstNode props 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="flex items-center justify-center  scale-x-[-1] first-node-class"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="red"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="w-full h-[6px] bg-background-accent-200 undefined"
  />
  <div
    class="flex items-center justify-center  undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with secondNode props 1`] = `
<div
  class="flex items-end flex-1 undefined"
>
  <div
    class="flex items-center justify-center  scale-x-[-1] undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="w-full h-[6px] bg-background-accent-200 undefined"
  />
  <div
    class="flex items-center justify-center  second-node-class"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="blue"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`FrameConnector > renders correctly with vertical prop 1`] = `
<div
  class="flex flex-col w-fit undefined"
>
  <div
    class="flex items-center justify-center rotate-90 origin-center scale-x-[-1] undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="h-full w-[6px] bg-background-accent-200 undefined"
  />
  <div
    class="flex items-center justify-center rotate-90 origin-center undefined"
  >
    <svg
      fill="none"
      viewBox="0 0 60 60"
      width="60"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M60 3.78963C60 1.56987 58.1925 -0.257305 55.9914 0.0298035C43.1529 1.70443 32.8081 11.3131 30.0386 23.7742C30.0336 23.7966 30 23.7935 30 23.7705V23.7705V23.7705C30 23.8917 29.9864 24.0125 29.9615 24.1312C29.5844 25.9293 29.3647 27.7853 29.3174 29.6841C29.3088 30.029 29.2729 30.374 29.1934 30.7097C26.0622 43.9325 14.18 53.7705 -2.67029e-05 53.7705H0.0999985C0.0447693 53.7705 0 53.8152 0 53.8705V59.6705C0 59.7257 0.0447731 59.7705 0.0999985 59.7705H52.9C52.9013 59.7705 52.9025 59.7704 52.9037 59.7704H55C57.7614 59.7704 60 57.5318 60 54.7704V3.78963Z"
        fill="#d0cac4"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
